### PR TITLE
Stats: Swapping Posts & pages component

### PR DESCRIPTION
**Pull request by grzegorz-cp**
#### Proposed Changes

* adding a feature flag and replacing `Posts & pages` StatsModule with the recently created component with horizontal bars

#### Testing Instructions

* Navigate to the main `Stats` page either on a local instance or the test URL from the comment below

Before
Screenshot 2022-11-02 at 12 16 24 PM

After
Screenshot 2022-11-02 at 12 17 38 PM

#### Pre-merge Checklist

- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [x] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?

Related to #69044

### CODE REVIEW

File: client/my-sites/stats/stats-list/stats-list-posts.jsx [hunk 1]
```diff
@@ -0,0 +1,133 @@
+import {
+	// eslint-disable-next-line import/named
+	HorizontalBarList,
```
Comment (grzegorz-cp) [inline client/my-sites/stats/stats-list/stats-list-posts.jsx]: I'm not sure why isn't this recognised. Lint doesn't see it but the exports are there...

Comment (jsnmoon) [reply client/my-sites/stats/stats-list/stats-list-posts.jsx]: It works fine for me without the `eslint` override comments on my local machine 🤔

Comment (grzegorz-cp) [reply client/my-sites/stats/stats-list/stats-list-posts.jsx]: I will double-check that. Maybe it's just my linter or IDE setup.

Comment (grzegorz-cp) [reply client/my-sites/stats/stats-list/stats-list-posts.jsx]: Downgrading `typescript` from 4.7.4 to 4.7.2 resolved the issue but since 4.7.4 is in `package.json` it's not a great solution... What version are you using?

File: client/my-sites/stats/stats-list/stats-list-actions.jsx [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+			switch ( action.type ) {
+				case 'follow':
+					if ( action.data ) {
+						actionItem = (
+							<Follow
+								key={ action.type }
+								moduleName={ moduleNameTitle }
+								isFollowing={ !! action.data.is_following }
+								siteId={ action.data.blog_id }
+								translate={ translate }
```
Comment (grzegorz-cp) [inline client/my-sites/stats/stats-list/stats-list-actions.jsx]: Do I need to wrap the components with `localize()` even if I'm only passing the method down?

File: client/my-sites/stats/stats-list/stats-list-posts.jsx (hunk 1 above)
Comment (jsnmoon) [reply client/my-sites/stats/stats-list/stats-list-posts.jsx]: Strange, I'm using 4.7.4 as well but not seeing any errors.

**Comment by jsnmoon** [comment]
Pushed a trunk merge in bbf8421.

File: client/my-sites/stats/stats-list/stats-list-actions.jsx (hunk 2 above)
Comment (jsnmoon) [reply client/my-sites/stats/stats-list/stats-list-actions.jsx]: I think it'd be better to wrap the component in `localize()` or the `useTranslate` hook instead of passing the method down as a property.

If this isn't possible, then no. No need to wrap the component in `localize()` if it already has the `translate` function available.

File: client/my-sites/stats/stats-list/stats-list-actions.jsx [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+					postId={ data.id }
+					key={ 'promote-post-' + data.id }
+					moduleName={ moduleNameTitle }
+					onToggleVisibility={ () => {} } // obsolete function that was blocing a general onClick handler from publishing unrelated GA events
+				/>
+			);
+		}
+
+		if ( actionItems.length > 0 ) {
+			actionList = (
```
Comment (jsnmoon) [inline client/my-sites/stats/stats-list/stats-list-actions.jsx]: I don't think we need to instantiate `actionList` at all if we just return the node here:

```suggestion
			return (
```

File: client/my-sites/stats/stats-list/stats-list-actions.jsx [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+			}
+		} );
+
+		if ( moduleName === 'posts' && data.public ) {
+			actionItems.push(
+				<Promote
+					postId={ data.id }
+					key={ 'promote-post-' + data.id }
+					moduleName={ moduleNameTitle }
+					onToggleVisibility={ () => {} } // obsolete function that was blocing a general onClick handler from publishing unrelated GA events
```
Comment (jsnmoon) [inline client/my-sites/stats/stats-list/stats-list-actions.jsx]:
```suggestion
					onToggleVisibility={ () => {} } // obsolete function that was blocking a general onClick handler from publishing unrelated GA events
```

File: client/my-sites/stats/stats-list/stats-list-posts.jsx [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+	// error,
+	loader,
+	translate,
+} ) => {
+	// const [ activeGroups, setActiveGroup ] = useState( [] );
+	const moduleNameTitle = titlecase( moduleName );
+
+	// const isGroupActive = ( groupName ) => {
+	// 	return activeGroups.indexOf( groupName ) >= 0;
+	// };
```
Comment (jsnmoon) [inline client/my-sites/stats/stats-list/stats-list-posts.jsx]: Could we get rid of these code comments if they're no longer necessary?

File: client/my-sites/stats/stats-list/stats-list-posts.jsx [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+
+	const localClickHandler = ( event, listItemData ) => {
+		debug( 'clickHandler' );
+
+		page( listItemData?.page );
+
+		gaRecordEvent( 'Stats', ` Clicked ${ moduleNameTitle } Summary Link in List` );
```
Comment (jsnmoon) [inline client/my-sites/stats/stats-list/stats-list-posts.jsx]: Nitty suggestion, feel free to ignore:

```suggestion
		debug( 'clickHandler' );
		page( listItemData?.page );
		gaRecordEvent( 'Stats', ` Clicked ${ moduleNameTitle } Summary Link in List` );
```

File: client/my-sites/stats/stats-module/index.jsx [hunk 7]
```diff
@@ -152,66 +154,103 @@ class StatsModule extends Component {
...
-					{ statType === 'statsFileDownloads' && (
-						<StatsModuleAvailabilityWarning
-							statType={ statType }
-							startOfPeriod={ period && period.startOf }
-						/>
+
+				<div className={ classNames( { 'old-stats-module-hidden': shouldHideOldModule } ) }>
+					{ siteId && statType && (
+						<QuerySiteStats statType={ statType } siteId={ siteId } query={ query } />
 					) }
```
Comment (jsnmoon) [inline client/my-sites/stats/stats-module/index.jsx]: Should we avoid rendering the old stats module entirely?

```suggestion
				{ ! shouldHideOldModule && <div>
```

File: config/wpcalypso.json [hunk 8]
```diff
@@ -141,6 +141,7 @@
 		"site-indicator": true,
 		"ssr/prefetch-timebox": true,
 		"stats/new-main-chart": false,
+		"stats/new-stats-module-component": true,
```
Comment (jsnmoon) [inline config/wpcalypso.json]: Let's make sure to disable this before merging/shipping this PR.

File: client/my-sites/stats/stats-list/stats-list-actions.jsx [hunk 9]
```diff
@@ -0,0 +1,102 @@
+import classNames from 'classnames';
+import React, { useState } from 'react';
+import titlecase from 'to-title-case';
+import Follow from './action-follow';
+import OpenLink from './action-link';
+import Page from './action-page';
+import Promote from './action-promote';
+import Spam from './action-spam';
+
+const StatsListActions = ( { data, moduleName, translate } ) => {
```
Comment (jsnmoon) [inline client/my-sites/stats/stats-list/stats-list-actions.jsx]: Could we add a brief comment here explaining what this component does?

File: client/my-sites/stats/stats-module/index.jsx [hunk 10]
```diff
@@ -152,66 +154,103 @@ class StatsModule extends Component {
...
-			<div>
-				{ siteId && statType && (
-					<QuerySiteStats statType={ statType } siteId={ siteId } query={ query } />
-				) }
-				{ ! isAllTime && (
-					<SectionHeader
-						className={ headerClass }
-						label={ this.getModuleLabel() }
-						href={ ! summary ? summaryLink : null }
+			<>
```
Comment (jsnmoon) [inline client/my-sites/stats/stats-module/index.jsx:160]: The changes here are hard to parse due to git butchering the diff, unfortunately.

Relocating `QuerySiteStats` here may make the diff easier to read:

```suggestion
			<>
				{ siteId && statType && (
					<QuerySiteStats statType={ statType } siteId={ siteId } query={ query } />
				) }
```

File: client/my-sites/stats/stats-list/stats-list-actions.jsx (hunk 2 above)
Comment (grzegorz-cp) [reply client/my-sites/stats/stats-list/stats-list-actions.jsx]: Thanks. All of the components are wrapped in localize so I can drop the props here.

File: client/my-sites/stats/stats-module/index.jsx (hunk 10 above)
Comment (grzegorz-cp) [reply client/my-sites/stats/stats-module/index.jsx:160]: I moved it manually, thanks for the suggestions. Let me know if this helps.

Comment (jsnmoon) [reply client/my-sites/stats/stats-module/index.jsx:160]: Helps a ton, thanks!

**Review by jsnmoon — APPROVED** [review]
Looks good! Ship at your leisure.
